src/MessagePack.UnityClient/Assets/Scripts/Tests/ShareTests/ContractlessStandardResolverTest.cs

```diff
@@ -220,5 +244,27 @@ public void LongestStringCheck()
 
             v.IsStructuralEqual(o);
         }
+
+        [Fact]
+        public void NewFieldCheck()
+        {
+            var o = new NewField { X = "Foo", Y = "Bar" };
+            var bin = MessagePackSerializer.Serialize(o, Resolvers.ContractlessStandardResolver.Options);
+            NewField v =
+                MessagePackSerializer.Deserialize<NewField>(bin, Resolvers.ContractlessStandardResolver.Options);
+
+            v.IsStructuralEqual(o);
```

Comment:
I would want to see this test expanded to assert what happens to the base properties' values as well. e.g. are they never serialized? Are they deserialized or left at their default values?
If you drop the base type properties' values, why are they less deserving of serialization than the derived type? Those values are still accessible and may matter to someone.

My proposal: give each **new** member's name a qualifier that is the type they were defined on. This would allow maintaining storage compatibility with the base type since the property name for that member would not have changed, while allowing the new member's value to be stored.

Reply:
Good idea, I changed the code to preserve the value of the base member. Since a property can hide a field or vice versa, I also consolidated some of the duplicate code that existed for properties and fields.